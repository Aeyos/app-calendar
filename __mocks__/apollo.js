import { card, event } from './';

export default {
  allCards: {
    edges: [
      {
        node: { ...card },
      },
    ],
  },
  events: [event],
  loading: false,
  refetch: jest.fn(),
};
