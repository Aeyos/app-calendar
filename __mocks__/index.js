import MockDate from 'mockdate';

MockDate.set('2017-12-01');

export const card = {
  due_date: '2017-12-15T16:00:00-02:00',
  id: '1',
  title: 'Testing card',
};

export const event = {
  end: new Date(new Date(card.due_date).getTime() + 30 * 60000),
  id: card.id,
  start: new Date(card.due_date),
  title: card.title,
};

export const e = { preventDefault: jest.fn() };

export const pipefy = {
  app: { pipeId: '1' },
  locale: 'en',
  openCard: jest.fn(),
};
