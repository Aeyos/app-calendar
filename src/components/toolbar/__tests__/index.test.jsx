import React from 'react';
import renderer from 'react-test-renderer';

import Toolbar from '../';
import * as bigCalendarProps from '../../../../__mocks__/big_calendar';

describe('Toolbar', () => {
  it('render with a truthy loading props', () => {
    const component = renderer.create(<Toolbar {...bigCalendarProps} loading />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('render with a falsy loading props', () => {
    const component = renderer.create(<Toolbar {...bigCalendarProps} loading={false} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
