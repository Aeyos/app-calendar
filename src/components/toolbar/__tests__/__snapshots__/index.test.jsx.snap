// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar render with a falsy loading props 1`] = `
<div
  className="rbc-toolbar"
>
  <span
    className="rbc-toolbar-label"
  >
    Testing label
  </span>
  <span
    className="rbc-btn-group pipe-navigate"
  >
    <a
      className="prev"
      href="#navigate_prev"
      onClick={[Function]}
    >
      <span
        className="pp-ico-bold-arrow-left"
      />
    </a>
    <a
      className="today"
      href="#navigate_today"
      onClick={[Function]}
    >
      Today
    </a>
    <a
      className="next"
      href="#navigate_next"
      onClick={[Function]}
    >
      <span
        className="pp-ico-bold-arrow-right"
      />
    </a>
  </span>
  <span
    className="rbc-btn-group"
  >
    <button
      className="rbc-active"
      onClick={[Function]}
    >
      month
    </button>
    <button
      className=""
      onClick={[Function]}
    >
      week
    </button>
    <button
      className=""
      onClick={[Function]}
    >
      day
    </button>
    <button
      className=""
      onClick={[Function]}
    >
      agenda
    </button>
  </span>
</div>
`;

exports[`Toolbar render with a truthy loading props 1`] = `
<div
  className="rbc-toolbar"
>
  <span
    className="rbc-toolbar-label"
  >
    Testing label
  </span>
  <span
    className="rbc-btn-group pipe-navigate"
  >
    <span
      className="pp-ico-loading pp-color-info"
    />
  </span>
</div>
`;
